### PR TITLE
PACT-617 and PACT-896

### PR DESCRIPTION
**Pull request by rwalpole**
Creating interfaces/traits for database access and connecting the List Asset Legal Status Summaries service to the database.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/omega/api/repository/OmegaRepository.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import uk.gov.nationalarchives.omega.api.models.LegalStatus
+
+import scala.util.Try
+
+class OmegaRepository(sparqlConnector: SparqlEndpointConnector) extends AbstractRepository {
+
+  private val sparqlResourceDir = "sparql"
+  private val getLegalStatusSummarySparqlResource = s"/$sparqlResourceDir/select-legal-status-summaries.rq"
+  // private val getLegalStatusSummarySparqlResource = s"/select-legal-status-summaries.rq"
+
```
Comment (userjd) [inline src/main/scala/uk/gov/nationalarchives/omega/api/repository/OmegaRepository.scala:35]: can this be removed?

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/legalstatus/LegalStatusService.scala [hunk 2]
```diff
@@ -25,11 +25,18 @@ import cats.data.Validated
 import uk.gov.nationalarchives.omega.api.business.{ BusinessRequestValidation, BusinessService, BusinessServiceError, BusinessServiceReply, BusinessServiceRequest }
 import uk.gov.nationalarchives.omega.api.messages.StubData
 import io.circe.syntax.EncoderOps
+import uk.gov.nationalarchives.omega.api.repository.{ AbstractRepository, OmegaRepository }
```
Comment (userjd) [inline src/main/scala/uk/gov/nationalarchives/omega/api/business/legalstatus/LegalStatusService.scala:28]: `OmegaRepository` import unused

File: src/main/scala/uk/gov/nationalarchives/omega/api/services/ApiService.scala [hunk 3]
```diff
@@ -124,13 +128,14 @@ class ApiService(val config: ServiceConfig) extends Stateful {
     consumer: JmsAcknowledgerConsumer[IO],
     producer: JmsProducer[IO],
     localMessageStore: LocalMessageStore,
+    repository: OmegaRepository,
```
Comment (userjd) [inline src/main/scala/uk/gov/nationalarchives/omega/api/services/ApiService.scala:131]: For consistency could this be renamed to `omegaRepository` as above?

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/legalstatus/LegalStatusService.scala (hunk 2 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/business/legalstatus/LegalStatusService.scala]: Removed.

File: src/main/scala/uk/gov/nationalarchives/omega/api/services/ApiService.scala (hunk 3 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/services/ApiService.scala]: Done.

**Review by userjd — APPROVED** [review]

**Pull request merged**